src/rdb.c

```diff
@@ -2449,31 +2449,28 @@ void backgroundSaveDoneHandlerDisk(int exitcode, int bysignal) {
         if (bysignal != SIGUSR1)
             server.lastbgsave_status = C_ERR;
     }
-    server.rdb_child_pid = -1;
-    server.rdb_child_type = RDB_CHILD_TYPE_NONE;
-    server.rdb_save_time_last = time(NULL)-server.rdb_save_time_start;
-    server.rdb_save_time_start = -1;
-    /* Possibly there are slaves waiting for a BGSAVE in order to be served
-     * (the first stage of SYNC is a bulk transfer of dump.rdb) */
-    updateSlavesWaitingBgsave((!bysignal && exitcode == 0) ? C_OK : C_ERR, RDB_CHILD_TYPE_DISK);
 }
 
 /* A background saving child (BGSAVE) terminated its work. Handle this.
  * This function covers the case of RDB -> Slaves socket transfers for
  * diskless replication. */
-void backgroundSaveDoneHandlerSocket(int exitcode, int bysignal) {
+static void backgroundSaveDoneHandlerSocket(int exitcode, int bysignal) {
     if (!bysignal && exitcode == 0) {
         serverLog(LL_NOTICE,
             "Background RDB transfer terminated with success");
+        server.lastsave = time(NULL);
+        server.lastbgsave_status = C_OK;
```

Comment:
@trevor211 setting these two here would damage the mechanism in `serverCron` that decides when to trigger another persistence RDB save, and the mechanism in `writeCommandsDeniedByDiskError`.
i.e. the sockets save doesn't generate an persistence rdb file.

regarding the INFO fields, although they don't affect redis, some orchestration framework that monitors redis may use them to conclude things about persistence state, so i'm not sure we wanna change them.

i.e. for each of these info fields, we need to understand if it's use is mainly to monitor the state of the persistence, or gain knowledge about and infrastructure and a recent `fork` (regardless of it's purpose).

Reply:
Didn't notice that, I'll fix it just keeping the `rdb_last_bgsave_time_sec ` part.

Reply:
the refactoring to reduce duplicate code seems good.

But i'm not even sure that we wanna affect `rdb_last_bgsave_time_sec`.
looking at the other fields next to it, they all speak about persistence, and it does say "bgsave" in its name:
```c
            "rdb_changes_since_last_save:%lld\r\n"
            "rdb_bgsave_in_progress:%d\r\n"
            "rdb_last_save_time:%jd\r\n"
            "rdb_last_bgsave_status:%s\r\n"
            "rdb_last_bgsave_time_sec:%jd\r\n"
            "rdb_current_bgsave_time_sec:%jd\r\n"
```
and that's how it's documented in https://redis.io/commands/info
i suppose i wouldn't mind to update the documentation, and that it won't break any application, but i don't like the field name being misleading.

is there any reason you think this change is important?

Reply:
I noticed that `rdb_bgsave_in_progress` and `rdb_current_bgsave_time_sec` works for both socket based rdb and disk based rdb. So maybe `bgsave` means background saving for both.

I also noticed that `server.rdb_save_time_start` is set when a bgsave starts:
https://github.com/redis/redis/blob/04a0af9085ed91eace8af94f71a07b651a1af454/src/rdb.c#L1412
https://github.com/redis/redis/blob/04a0af9085ed91eace8af94f71a07b651a1af454/src/rdb.c#L2621

and it is reset to -1 when bgsave ends:
https://github.com/redis/redis/blob/04a0af9085ed91eace8af94f71a07b651a1af454/src/rdb.c#L2455
https://github.com/redis/redis/blob/04a0af9085ed91eace8af94f71a07b651a1af454/src/rdb.c#L2476

and there is a difference when bgsave ends about updating `server.rdb_save_time_last`:
https://github.com/redis/redis/blob/04a0af9085ed91etace8af94f71a07b651a1af454/src/rdb.c#L2454
(No update for `server.rdb_save_time_last` in function backgroundSaveDoneHandlerSocket)

In my opinion, we should make it consistent for both cases.
Either let `bgsave` for merely disk based rdb, for which case, we could probaly change the field of the output of `INFO PERSISTENSE`,  or we let `bgsave` for both cases while some info fields remain specific to disk based rdb.
For the latter, I think we should emit the common fields in a common place. That's why I did the trivial refactory.

Reply:
@trevor211 you right (convinced me this is already breached).
so considering i don't wanna change the meaning of `rdb_bgsave_in_progress`, we can update `rdb_last_bgsave_time_sec` too.
so this PR can be merged in it's current form, right?

Reply:
Yes.